tnt/src/main/java/org/dcsa/conformance/standards/tnt/checks/TntChecks.java

```diff
@@ -116,36 +151,42 @@ public static ActionCheck responseContentChecks(
                         .map(String::trim)
                         .collect(Collectors.toSet());
 
-                Set<String> errors =
-                    filterNodesByEventType(body, SHIPMENT_EVENT_TYPE)
-                        .filter(
-                            node -> {
-                              JsonNode shipmentEventTypeCodeNode =
-                                  node.path("shipmentEventTypeCode");
-                              return !(shipmentEventTypeCodeNode.isMissingNode()
-                                      && shipmentEventTypeCodeNode.isNull())
-                                  && !expectedShipmentEventTypeCodes.contains(
-                                      shipmentEventTypeCodeNode.asText())
-                                  && !VALID_SHIPMENT_EVENT_TYPES.contains(
-                                      shipmentEventTypeCodeNode.asText());
-                            })
-                        .map(
-                            node ->
-                                "Missing or mismatched shipmentEventTypeCode "
-                                    + "for eventType SHIPMENT at "
-                                    + node.path("eventId").asText())
-                        .collect(Collectors.toSet());
-
-                return errors.isEmpty() ? Set.of() : errors;
+                Stream<JsonNode> filteredStream =
+                    filterNodesByEventType(eventNodes, SHIPMENT_EVENT_TYPE);
+                if (filteredStream.anyMatch(node -> true)) {
+                  issues.addAll(
+                      filterNodesByEventType(eventNodes, SHIPMENT_EVENT_TYPE)
+                          .filter(
+                              node -> {
+                                JsonNode shipmentEventTypeCodeNode =
+                                    node.path("shipmentEventTypeCode");
+                                return !isEmptyNode(shipmentEventTypeCodeNode)
```

Comment:
Isn't the shipmentEventTypeCode a mandatory attribute in shipmentEvent? Did the schema validation failed when that is removed?

Reply:
Its a required field and It will be checked as a part of the schema validation